extensions/common/aws/aws-s3-core/src/main/java/org/eclipse/edc/aws/s3/S3BucketSchema.java

```diff
@@ -19,11 +19,10 @@ public interface S3BucketSchema {
     String TYPE = "AmazonS3";
     String REGION = "region";
     String BUCKET_NAME = "bucketName";
-    String KEY_NAME = "keyName";
-    String KEY_PREFIX = "keyPrefix";
     String FOLDER_NAME = "folderName";
     String ACCESS_KEY_ID = "accessKeyId";
     String SECRET_ACCESS_KEY = "secretAccessKey";
     String ENDPOINT_OVERRIDE = "endpointOverride";
-    String SECRET_TOKEN = "secretToken";
+    String OBJECT_PREFIX = "objectPrefix";
+    String OBJECT_NAME = "objectName";
```

Comment:
please consider implementing a `Validator<DataAddress>` for `AmazonS3` types to be registered on the `DataAddressValidatorRegistry` that can ensure the correct `DataAddress` structure both source and destination.

in the upstream there are `HttpDataDataAddressValidator` and `KafkaDataAddressValidator` as examples

Reply:
> ~please consider implementing a `Validator<DataAddress>` for `AmazonS3` types to be registered on the `DataAddressValidatorRegistry` that can ensure the correct `DataAddress` structure both source and destination.~
> 
> ~in the upstream there are `HttpDataDataAddressValidator` and `KafkaDataAddressValidator` as examples~
> 
> Edit: the validator already exists, please update it accordingly

I added conditional nullability validation for the new properties to the existing validator.